src/matter/session/secure/CaseServer.ts

```diff
@@ -45,20 +45,23 @@ export class CaseServer implements ProtocolHandler<MatterDevice> {
         // Try to resume a previous session
         const resumptionId = Crypto.getRandomData(16);
         let resumptionRecord;
+
+        // We try to resume the session
         if (peerResumptionId !== undefined && peerResumeMic !== undefined && (resumptionRecord = server.findResumptionRecordById(peerResumptionId)) !== undefined) {
             const { sharedSecret, fabric, peerNodeId } = resumptionRecord;
             const peerResumeKey = await Crypto.hkdf(sharedSecret, ByteArray.concat(peerRandom, peerResumptionId), KDFSR1_KEY_INFO);
             Crypto.decrypt(peerResumeKey, peerResumeMic, RESUME1_MIC_NONCE);
 
+            // All good! Create secure session
+            const secureSessionSalt = ByteArray.concat(peerRandom, peerResumptionId);
+            const secureSession = await server.createSecureSession(sessionId, fabric, peerNodeId, peerSessionId, sharedSecret, secureSessionSalt, false, true, mrpParams?.idleRetransTimeoutMs, mrpParams?.activeRetransTimeoutMs);
+
             // Generate sigma 2 resume
             const resumeSalt = ByteArray.concat(peerRandom, resumptionId);
             const resumeKey = await Crypto.hkdf(sharedSecret, resumeSalt, KDFSR2_KEY_INFO);
             const resumeMic = Crypto.encrypt(resumeKey, new ByteArray(0), RESUME2_MIC_NONCE);
             await messenger.sendSigma2Resume({ resumptionId, resumeMic, sessionId });
```

Comment:
Add a try / catch to destroy session in case of error.
Check if the same should be done in Pase

Reply:
done